include/net/lwm2m.h

```diff
@@ -371,6 +375,16 @@ void lwm2m_firmware_set_write_cb(lwm2m_engine_set_data_cb_t cb);
  */
 lwm2m_engine_set_data_cb_t lwm2m_firmware_get_write_cb(void);
 
+/**
+ * @brief Set callback to get update state changed.
+ *
+ * LwM2M clients use this function to know the state of a firmware update
+ * process.
+ *
+ * @param[in] cb A callback function to receive the update state changes
+ */
+void lwm2m_register_update_state_callback(lwm2m_engine_update_state_cb cb);
```

Comment:
Wouldn't it be simpler to use existing mechanisms to monitor resource value (i.e. register pre/post_write callback) instead of introducing another specialized callback type?